File: 7KYU/Collatz Conjecture Length/solution.c
#include <stddef.h>

size_t collatz(unsigned long long int n) {

    size_t res = 1;
  
    while(n != 1){
      res = res + 1;
      if(n % 2 == 0) {
        n /= 2;
      }
      else {
        n = n * 3 + 1;
      }
    }
  
    return res;

}

File: 7KYU/Collatz Conjecture Length/solution.go
package kata

func Collatz(n int) int {

	c := 1
	for n != 1 {
		if n&1 == 0 {
			n /= 2
		} else {
			n = n*3 + 1
		}
		c++
	}
	return c
}
